src/main/java/fr/moribus/imageonmap/commands/maptool/UpdateCommand.java

```diff
@@ -51,28 +52,97 @@
-
-        URL url;
+        //TODO fix the issue where to many quick usage of offlineNameFetch will return null
+        ArrayList<String> arguments = getArgs();
+        String warningMsg;
+        if (arguments.size() > 4) {
+            warningMsg = "Too many parameters!"
+                    + " Usage: /maptool update [player name]:<map name> <new url> [stretched|covered]";
+            warning(I.t(warningMsg));
+            return;
+        }
+        if (arguments.size() < 2) {
+            warningMsg =
+                    "Too few parameters! Usage: /maptool update [player name]:<map name> <new url> [stretched|covered]";
+            warning(I.t(warningMsg));
+            return;
+        }
+        final String playerName;
+        final String mapName;
+        final String url;
+        final String resize;
+        final Player playerSender;
+        Player playerSender1;
+        try {
+            playerSender1 = playerSender();
+        } catch (CommandException ignored) {
+            if (arguments.size() == 2) {
+                throwInvalidArgument(
+                        I.t("Usage: /maptool update [player name]:<map name> <new url> [stretched|covered]"));
+            }
+            playerSender1 = null;
+        }
+        playerSender = playerSender1;
 
-        if (args.length < 1) {
-            throwInvalidArgument(I.t("You must give an URL and a map name to update."));
+        //Sent by a non player and not enough arguments
+        if (arguments.size() == 2 && playerSender == null) {
+            PluginLogger.warning("Usage: /maptool update [player name]:<map name> <new url> [stretched|covered]");
+            return;
         }
-        if (args.length < 2) {
-            throwInvalidArgument(I.t("You must give a map name to update."));
+
+        if (arguments.size() == 2) {
+            resize = "";
+            playerName = playerSender.getName();
+            mapName = arguments.get(0);
+            url = arguments.get(1);
+        } else {
+            if (arguments.size() == 4) {
+                if (playerSender != null && !Permissions.UPDATEOTHER.grantedTo(playerSender)) {
+                    info(playerSender, I.t("You can't use this command"));
+                    return;
+                }
+                playerName = arguments.get(0);
+                mapName = arguments.get(1);
+                url = arguments.get(2);
+                resize = arguments.get(3);
+            } else {
+                if (arguments.size() == 3) {
+                    if (arguments.get(2).equals("covered") || arguments.get(2).equals("stretched")) {
+                        playerName = playerSender.getName();
+                        mapName = arguments.get(0);
+                        url = arguments.get(1);
+                        resize = arguments.get(2);
+                    } else {
+                        if (playerSender != null && !Permissions.UPDATEOTHER.grantedTo(playerSender)) {
```

Comment:
`grantedTo` accepte un `Permissible`, ce qu'est un `CommandSender`.
```suggestion
                        if (!Permissions.UPDATEOTHER.grantedTo(sender)) {
```